experimental/PlaneCurve/EllCurveZnZ.jl

```diff
@@ -621,7 +560,8 @@ function compute_ell_curve(N::fmpz, D::fmpz)
    elseif D == ZZ(-4)
       return (R.([-1, 0]), ZZ(1))
    else
-      p = hilbert_class_polynomial(D)
+      T, t = ZZ["t"]
+      p = hilbert_class_polynomial(T, D)
```

Comment:
```suggestion
      p = hilbert_class_polynomial(T, Int(D))
```

You might want to add some tests that reaches this: the function up there is
`hilbert_class_polynomial(R::FmpzPolyRing, d::Int)`
and your `D` here is an fmpz. You will have to somewhere cast this `D` as `Int(D)`. (This cast can fail, but if is does fail, the class polynomial for that D is not computable anyways.)